pkg/channel/distributed/controller/kafkachannel/dispatcher.go

```diff
@@ -144,20 +148,35 @@ func (r *Reconciler) reconcileDispatcherService(ctx context.Context, logger *zap
 		}
 	} else {
 
+		// Determine whether the existing service is different in a way that demands a patch
+		// such as missing required labels or spec differences
+		patch, needsUpdate := util.CheckServiceChanged(logger, existingService, newService)
+
 		// Log Deletion Timestamp & Finalizer State
-		if service.DeletionTimestamp.IsZero() {
+		if existingService.DeletionTimestamp.IsZero() {
 			logger.Info("Successfully Verified Dispatcher Service")
 		} else {
-			if util.HasFinalizer(r.finalizerName(), &service.ObjectMeta) {
+			if util.HasFinalizer(r.finalizerName(), &newService.ObjectMeta) {
 				logger.Info("Blocking Pending Deletion Of Dispatcher Service (Finalizer Detected)")
 			} else {
 				logger.Warn("Unable To Block Pending Deletion Of Dispatcher Service (Finalizer Missing)")
 			}
 		}
 
-		// Return Success
-		return nil
```

Comment:
I think the finalizer log statements above were relying on this common exit `return nil` and might not make sense without it?

Reply:
It's still returning nil, it's just after the "if needsUpdate" block